### PR TITLE
tests: debug flaky expect test

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Catchpoint test exists like algod process was killed, so added `top` invocation to monitor algod processes presence as well as machine memory usage.
Additionally fixed a wrong hardcoded wait round number in this test - it would not cause a failure but still a small inconsistency. 

## Test Plan

This is test-related code

### CODE REVIEW

File: test/e2e-go/cli/goal/expect/goalExpectCommon.exp [hunk 1]
```diff
@@ -52,6 +52,13 @@ proc ::AlgorandGoal::Abort { ERROR } {
         puts "GLOBAL_TEST_ROOT_DIR $::GLOBAL_TEST_ROOT_DIR"
         puts "GLOBAL_NETWORK_NAME $::GLOBAL_NETWORK_NAME"
 
+        # dump system and processes into to check memory consumption and if algod procs are still alive
+        if { $tcl_platform(os) == "Darwin" } {
```
Comment (algonautshant) [inline test/e2e-go/cli/goal/expect/goalExpectCommon.exp:56]: Do these tests run on windows?

File: test/e2e-go/cli/goal/expect/goalExpectCommon.exp [hunk 2]
```diff
@@ -52,6 +52,13 @@ proc ::AlgorandGoal::Abort { ERROR } {
         puts "GLOBAL_TEST_ROOT_DIR $::GLOBAL_TEST_ROOT_DIR"
         puts "GLOBAL_NETWORK_NAME $::GLOBAL_NETWORK_NAME"
 
+        # dump system and processes into to check memory consumption and if algod procs are still alive
```
Comment (algonautshant) [inline test/e2e-go/cli/goal/expect/goalExpectCommon.exp:55]: Typo
```suggestion
        # dump system and processes to check memory consumption and if algod procs are still alive
```

**Review by algonautshant — APPROVED** [review]
Looks good. 
Couple of minor comments.

File: test/e2e-go/cli/goal/expect/goalExpectCommon.exp (hunk 1 above)
Comment (algorandskiy) [reply test/e2e-go/cli/goal/expect/goalExpectCommon.exp:56]: I do not think so but added Linux case to be exact

File: test/e2e-go/cli/goal/expect/goalExpectCommon.exp (hunk 2 above)
Comment (algorandskiy) [reply test/e2e-go/cli/goal/expect/goalExpectCommon.exp]: fixed

**Review by algonautshant — APPROVED** [review]

**Pull request merged**